### ISSUE TITLE
Theme preview missing

### ISSUE THREAD

**Issue by scottfurry** [eclipse-color-theme/eclipse-color-theme#213]
I recently switch to a gentoo-based install and was putting my build environment back together.
I grabbed the latest Eclipse Color Theme from the github update site. It works wonderfully. THX

However, I noticed that in the preferences screen that the theme preview is missing.
I'm hoping that I didn't miss some package (possible with src-based distribution). Is It possible there's a bug in the latest package?

Screen shot of preference screen
![eclipsecolortheme_nopreview](https://cloud.githubusercontent.com/assets/6686571/4855416/ca33b840-60a4-11e4-930d-6299ee20ee90.png)

**Comment by scottfurry** [comment]
tl:dr - swt support is a dog's breakfast. problem isn't with this project.

I was able to workaround the issue, but it was messy.
Ref: http://www-01.ibm.com/support/docview.wss?uid=swg21271865
http://www.eclipse.org/swt/faq.php#browserlinux
http://stackoverflow.com/a/24260503

For Funtoo (deriviative of Gentoo), I had to do the following...
- (as root) ensure swt was installed (dev-java/swt)
- (as root) cd /opt
- (as root) wget http://ftp.mozilla.org/pub/mozilla.org/xulrunner/releases/24.0/runtimes/xulrunner-24.0.en-US.linux-x86_64.tar.bz2
- (as root) tar -xjf xulrunner-*.bz2 (now have /opt/xulrunner folder)
- (as user) edit eclipse.ini with the following:

    -Dorg.eclipse.swt.browser.DefaultType=mozilla
    -Dorg.eclipse.swt.browser.XULRunnerPath=/opt/xulrunner

- (as user) create a start script for eclipse

     #!/bin/bash
     export MOZILLA_FIVE_HOME=/opt/xulrunner
     export LD_LIBRARY_PATH=${MOZILLA_FIVE_HOME}:${LD_LIBRARY_PATH}
     export ECLIPSE_HOME=[path/to/eclipse] <---(note path to folder not executable)
     "${ECLIPSE_HOME}/eclipse" $*

- (as user) mark script as executable
- (as user) start eclipse using the script

ColorTheme Preferences should now display preview of theme and you should not have output on stderr.

If you cannot get the latest version of swt from your distribution...
(steps below replaces first line above)
- download from here: http://download.eclipse.org/eclipse/downloads/drops4/R-4.4-201406061215/#SWT
- expand zip file to its own folder (I tried /home/user/bin/swt for example)
- edit eclipse ini and add "-Djava.library.path=/home/user/bin/swt"

YMMV
Good Luck

**Comment by fhd** [comment]
> tl:dr - swt support is a dog's breakfast. problem isn't with this project.

Closing then :smile:

Thanks for posting the workaround I suppose something along those lines should help a bunch of people with this problem.

**Comment by scottfurry** [comment]
Simpler Alternative is to install webkitgtk for GTK2 (see [Eclipse Bug - 424657](https://bugs.eclipse.org/bugs/show_bug.cgi?id=424657))
under Arch linux this is both webkitgtk and webkitgtk2.
webkitgtk2 was installed and preview non-functional.
Installed webkitgtk and preview works perfectly.
Again...YMMV
